expr/func_minmax_test.go

```diff
@@ -0,0 +1,235 @@
+package expr
+
+import (
+	"math"
+	"strconv"
+	"testing"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/schema"
+	"github.com/grafana/metrictank/test"
+)
+
+func getNewMinMax(in []models.Series) *FuncMinMax {
+	f := NewMinMax()
+	ps := f.(*FuncMinMax)
+	ps.in = NewMock(in)
+	return ps
+}
+
+const interval = 10
+
+var basic = []schema.Point{
+	{Val: 0, Ts: 10},
+	{Val: 10, Ts: 20},
+	{Val: 20, Ts: 30},
+}
+
+var basicR = []schema.Point{
+	{Val: 0, Ts: 10},
+	{Val: float64(10) / 20, Ts: 20},
+	{Val: 1, Ts: 30},
+}
+
+var nan = []schema.Point{
+	{Val: 0, Ts: 10},
+	{Val: math.NaN(), Ts: 20},
+	{Val: 20, Ts: 30},
+}
+
+var nanR = []schema.Point{
+	{Val: 0, Ts: 10},
+	{Val: math.NaN() / 20, Ts: 20},
+	{Val: 1, Ts: 30},
+}
+
+var minMaxSame = []schema.Point{
+	{Val: 20, Ts: 10},
+	{Val: 20, Ts: 20},
+	{Val: 20, Ts: 30},
+}
+
+var minMaxSameR = []schema.Point{
+	{Val: 0, Ts: 10},
+	{Val: 0, Ts: 20},
+	{Val: 0, Ts: 30},
+}
+
+var infinity = []schema.Point{
+	{Val: 100, Ts: 10},
+	{Val: 1000, Ts: 20},
+	{Val: math.Inf(0), Ts: 30},
+}
+
+// infinity / infinity is undefined
+var infinityR = []schema.Point{
+	{Val: 0, Ts: 10},
+	{Val: 0, Ts: 20},
+	{Val: math.NaN(), Ts: 30},
+}
+
```

Comment:
I think it's good to add a "0 input series" case (can happen after functions like removeEmptySeries) and a "2 input series" case (to make sure that handling of arrays is correct). I am going to go through the functions at some point and try to normalize the testing of them.

Reply:
Would it be better to add some of this test data to  `/expr/data_test.go`

There's one that I added for `removeEmptySeries` with all zeros that you could use if needed.
https://github.com/grafana/metrictank/blob/master/expr/data_test.go#L74-L81